### PR TITLE
Implementing Pagination and Filtering in Lists

### PR DESCRIPTION
**Pull request by ErikGmz**
# Description

- Installed and properly configured Pagy gem for Accommodations lists.
- Implemented prototype filters for accommodations searching.
- Added TomSelect library to improve multiple options selection controls.
- Added RangeSlider library to implement double sliders controls.
- Renamed 'name' column to 'title' in Accommodation model (this to prevent errors when joining with Details model).

## Examples

Now it's possible to search for accommodations based on the following features:
- Hosts.
- Categories.
- Details.
- Title.
- Prices per day range.
- Ratings range.
- Bedrooms numbers range.
- Bathrooms numbers range.
- Beds numbers range.
- Max guests numbers range.
- Availability dates range.
- Address.

### CODE REVIEW

File: app/javascript/controllers/rangeslider_controller.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  connect() {
+    let field = document.getElementById(this.fieldValue);
+
+    if (this.rangeValue !== undefined && this.rangeValue !== "") {
+      var range = this.rangeValue.split(",").map(number => parseInt(number)).sort((a, b) => a - b)
+    }
+    else {
+      var range = [this.minValue, this.maxValue]
+    }
```
Comment (MauricioNC) [inline app/javascript/controllers/rangeslider_controller.js:20]: What about to modify this to:

```javascript
let range = [this.minValue, this.maxValue]

if (this.rangeValue !== undefined && this.rangeValue !== "") {
  range = this.rangeValue.split(",").map(number => parseInt(number)).sort((a, b) => a - b)
}
```

File: app/views/accommodations/_search_filter_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      <%= form.text_field :address, value: params[:address],
+                          class: "block shadow rounded-md border border-gray-200 outline-none px-3 py-2 mt-2 w-72" %>
+    </div>
+  </div>
+
+  <div class='flex flex-col items-center min-[250px]:flex-row'>
+    <%= form.submit 'Search', class: 'rounded-lg py-3 px-5 bg-gray-100 inline-block font-medium' %>
+    <%= link_to 'Clear', accommodations_path, class: 'rounded-lg py-3 px-5 bg-gray-100 inline-block font-medium ml-3 no-underline' %>
+  </div>
+<% end %>
```
Comment (MauricioNC) [inline app/views/accommodations/_search_filter_form.html.erb:104]: Add a new line at the end of your code

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: app/javascript/controllers/rangeslider_controller.js (hunk 1 above)
Comment (ErikGmz) [reply app/javascript/controllers/rangeslider_controller.js]: Fixed!

File: app/views/accommodations/_search_filter_form.html.erb (hunk 2 above)
Comment (ErikGmz) [reply app/views/accommodations/_search_filter_form.html.erb]: Fixed!

**Review by MauricioNC — APPROVED** [review]

**Review by juperameza — APPROVED** [review]

**Review by STR1ND3R — APPROVED** [review]

**Review by DevLuis123 — APPROVED** [review]

**Pull request merged**